### DISCUSSION TITLE
labels in datasets used in ASReview evaluations.

### DISCUSSION THREAD

**Discussion by Rinus-Verdonschot-UB** [asreview/asreview#1927] [Q&A]
Dear forum

I have a clarification question, often I read in articles relating to simulations with ASReview that the labels are based on "Full-Text which are actually included in the final review". For example:

Ferdinands et al. (2023) -- "the labels in the data adhere to the researchers’ decision on which publications to include in the systematic review." (P3)

v/d Schoot et al. (2018)
![afbeelding](https://github.com/user-attachments/assets/c4adba4e-105b-4446-87ba-9433321fb259)

But, is this really how users would typically use the software? I would assume that people use ASReview to get to a larger "full-text articles assessed for eligibility" set and this set is then further scrutinized (by hand) into the final set appearing in the paper (without assistance of ASReview).

Most likely I am not understanding this correctly, but shouldn't the ASReview simulations then be based on the larger sample of articles (before this is then further scrutinized into the final set of actually included articles) as this is what people would typically use ASReview for. I could also imagine that a smaller (more homogeneous?) set would give more beneficial WSS and other values than a larger (perhaps more heterogeneous) set would. 

Could anybody clarify this point for me, as I am slightly puzzled what the typical/preferred set of labels should be when doing simulations with ASReview. 

Kind regards,

Rinus Verdonschot
University Library Maastricht

**Comment by jteijema** [comment]
That's a great question! We've touched on this before slightly, #1802, but not this in depth, thank you! I can answer for the optimization side of ASReview; we've used both TIAB and Fulltext datasets for the evaluation of our models. In our [scoping review](https://osf.io/preprints/psyarxiv/67zmt) we've seen 60 papers on the topic of active learning simulation studies. None of those mention the difference between TIAB and FullText datasets in relation to their performance. 

Personally, I think this is because:
1. The difference can be seen as noise, which models should be resistant to in some capacity.
2. It's VERY hard to find proper datasets. I think this stops a lot of research into this topic.

**Comment by Rensvandeschoot** [comment]
Thank you for your excellent question! It touches on a nuanced but important aspect of using ASReview for systematic reviews., so let me clarify :-) 

The main reason models in ASReview (and all other tools for screening prioritization) are optimized for identifying full-text relevant papers is related to the overarching goal of active learning-aided systematic reviews: systematically finding all truly relevant records as efficiently as possible from a large pool of documents to answer a specific research question.

To illustrate, imagine a total of _K_ records to be screened for relevance. Of these:
- _x_ is the subset screened using, for example ASReview,
- _y_ is the subset labeled as relevant based on the information available to the model and the screener (e.g., title/abstract or full text),
- _z_ is the subset of truly relevant records ised to answer the research question. at the end

The ideal scenario would be _x = y = z_, meaning the system prioritizes all and only truly relevant papers, saving maximum time and effort. A slightly less optimal—but still acceptable—outcome is when _y = z_ while _x_ is minimized, ensuring time savings through prioritization even if some irrelevant records are screened.

Why _x_ is not _z_ might be caused by a mismatch between the text presented to the model/human and the actual text with all the information. For a machine learning model to make correct predictions and a human to make a labeling decision, it relies on the input text provided. If the decision to include or exclude a record requires full-text information but only titles and abstracts are available, both humans and the model might mis-label/rank the record. This often happens when critical details are absent from the abstract but present in the full text. By contrast, when full-text information is used in the screening tool, the model/human has access to all the relevant details, leading to better alignment between _y_ and _z_.

Consider a simulation study. In the plot below, the X-axis represents the first 20% of records screened, while the Y-axis shows the percentage of title/abstract inclusions identified. At 15% of screening, 60% of title/abstract inclusions have been found. Red dots indicate full-text relevant papers. By 11% screening, all full-text relevant papers have been identified, allowing screening to stop—if the screener knows this. However, because full-text information is not (always) accessible during the abstract phase, the performance is sub-optimal. 

From a machine learning perspective, it makes sense to directly make full-text inclusion decisions and this is why we offer a hyperlink to the full-text (if the DOI or an URL is available in the data imported). This approach ensures that the system mimics the final decision-making process as closely as possible. However, practical constraints often limit access to full-text data (e.g., copyright restrictions, costs). 

So, while ASReview is often used for title/abstract screening as part of a multi-phase pipeline, its broader applicability extends to scenarios where full-text information is available and used in the software directly. Regardless of the input type, the models are optimized to achieve the underlying task: identifying all relevant records quickly and accurately. 

![bollenplot](https://github.com/user-attachments/assets/8c470370-a219-4280-88b9-123454452db8)

**Comment by Rinus-Verdonschot-UB** [comment]
Hi Jelle and Rens,

Thanks for your (elaborate) answers. To be honest, I am not sure understand your answers, probably because I asked the question wrongly. What I think is that people will use ASReview to make a smaller pile from the initial big pile (so, lets say from 6000 articles make a pile of lets say 100-200 articles which then are fully read and finally go into the paper or not). So, maybe 25 papers make the cut from this smaller pile.

So, the order would be big pile (N=6000) -> Asreview -> smaller pile (N=200) --> manual full text reading (no ASReview assistance) -> final Pile = 25 left (used in review).

Many of the simulations I read simulate with the latter (final pile). My question is, is this the pile we want to simulate on?
For example, I have a case in which the big pile (N=6383) to smaller pile yielded 327 articles which are then all fully read (so no help from ASReview would be needed) for a final decision to be put in the pile. That simulation looks like this:

![afbeelding](https://github.com/user-attachments/assets/1a5906a1-9bc7-4dad-8f61-0e9e11048a1e)

Eventually, from the smaller pile (N=327) after thorough full-text screening it was decided that 27 papers would make it to the final review paper.

If I only use those papers (N=27) for the simulation then we get this:

![afbeelding](https://github.com/user-attachments/assets/ca3d8ece-f7b7-4c1d-a9cd-dbaebcfafb41)

My question is (which you may have already tried to answer, but I may perhaps not have been clever enough to understand yet) is which simulation would be most useful to the likely user of ASReview.

I see many simulations out there which use the N=27 final pile, but I think the actual situation which people use ASReview in, is the intermediate/smaller pile (N=327 in this particular case). 

Also, if only simulations are ran with these N=27 then those 300 papers do not exert their influence on the pile ordering (as they are now deemed "not included", right?) which might obscure the simulation.

These were my questions/concerns. Once again, I might have completely misunderstood the whole matter, so I would be grateful if you could clarify it again with my specific example, which "pile" (N=327 or N=27) should I use in my simulations and why?

Thank you so much for your continued support and for making a great piece of software.

Kind regards,

Rinus Verdonschot
University Library Maastricht

**Reply by Rensvandeschoot** [reply]
I would use the n=27 pile, because the task to answer the research question is to identify the 27 records in the pile of 6383 records, and not to identify the 327 records. Let me clarify further based on your example.

In your scenario, the primary task of systematic reviewing is to identify the 27 truly relevant records from the initial pile of 6383 records. This means the goal of the ASReview simulations should be to optimize for identifying those 27 records as efficiently as possible. Stated differently, the ultimate goal of ASReview is to help researchers find the records that answer the research question—in this case, the 27 papers. 

The smaller pile of 327 records is just an intermediate result, often produced as part of a two-phase workflow (e.g., title/abstract screening followed by full-text screening). While this step reduces the workload, it is not the final outcome. Using this intermediate set for simulations could lead to models being optimized for a different task—identifying intermediate results rather than final inclusions. Moreover, including the 300 records (327 - 27) in simulations as "relevant" may dilute the signal, as these records are not the true target. Their presence could introduce noise that doesn’t align with the end goal of finding the 27 relevant records. This is especially important for training and evaluating machine learning models, as the model might learn to prioritize records that are not ultimately relevant.

Therefore and if possible, I would recommend integrating full-text decision making into ASReview directly (so screen an article, if relevant check full-text and make this decission in ASReview - then you optimally benifit from the performance of the machine learning models. I also realize this process might deviate from your normal workflow...

**Reply by Rinus-Verdonschot-UB** [reply]
Dear Rens, 

You are of course correct that the eventual task is to identify the N=27 articles, but what people will (likely) use ASReview for is to get from N=6000 to N=300 then to get to the final included this is typically done manually, but, if I understood it correctly, this can again be done via ASReview (incl. full-text, and not only tiab's). In any case, I thought that this (6000>300) might be a useful simulation to report, but I understood your point now why N=27 is preferred.

In any case, thanks a lot for your quick and clear reply and the FT suggestion. I will discuss this in my team. 

Kind regards, 

Rinus

**Reply by Rensvandeschoot** [reply]
I think a simulation showing the two different steps and explaining the difference is very useful! Your plots are very insightful!! The users of ASReview should not expect the second performance if they use the first strategy. So, it makes sense to align the expectations for using the software with the actual workflow someone is using.
